sql/log.cc

```diff
@@ -6099,8 +6088,13 @@ MYSQL_BIN_LOG::write_gtid_event(THD *thd, bool standalone,
   DBUG_ASSERT(this == &mysql_bin_log);
 
 #ifdef WITH_WSREP
-  if (wsrep_gtid_mode && is_gtid_cached(thd))
-    DBUG_RETURN(false);
+  if (wsrep_gtid_mode)
+  {
```

Comment:
Why these assignments are needed? These affect also non-wsrep threads if the `wsrep_gtid_mode` is on and might have some unwanted side effects.

Reply:
If we look at test: galera.mdev_10518, on a same thread we are creating also table with engine=MEMORY.
Later we try to insert something in that table which is not replicated, we should reset back to "original" (domain_id, server_id) after each gtid write so that non-replicated events have local tuple of domain_id and server_id.